Sources/System/FilePath/FilePathTemp.swift

```diff
@@ -0,0 +1,281 @@
+      return FilePath(SystemString(platformString: buffer.baseAddress!))
+    }
+    if let path = path {
+      return path
+    }
+  }
+  #else
+  return "/tmp"
+  #endif
+}
+
+internal func recursiveRemove(at path: FilePath) throws {
+  let dirfd = try FileDescriptor.open(path, .readOnly, options: .directory)
+  defer {
+    try? dirfd.close()
+  }
+
+  let dot: (CInterop.PlatformChar, CInterop.PlatformChar) = (46, 0)
+  try withUnsafeBytes(of: dot) {
+    try recursiveRemove(
+      in: dirfd.rawValue,
+      path: $0.assumingMemoryBound(to: CInterop.PlatformChar.self).baseAddress!
+    )
+  }
+
+  try path.withPlatformString {
+    if system_rmdir($0) != 0 {
+      throw Errno.current
+    }
+  }
+}
+
+fileprivate func impl_opendirat(
+  _ dirfd: CInt,
+  _ path: UnsafePointer<CInterop.PlatformChar>
+) -> UnsafeMutablePointer<system_DIR>? {
+  let fd = system_openat(dirfd, path,
+                         FileDescriptor.AccessMode.readOnly.rawValue
+                           | FileDescriptor.OpenOptions.directory.rawValue)
+  if fd < 0 {
+    return nil
+  }
+  return system_fdopendir(fd)
+}
+
+internal func forEachFile(
+  in dirfd: CInt, path: UnsafePointer<CInterop.PlatformChar>,
+  _ body: (system_dirent) throws -> ()
+) throws {
+  guard let dir = impl_opendirat(dirfd, path) else {
+    throw Errno.current
+  }
+  defer {
+    _ = system_closedir(dir)
+  }
+
+  while let dirent = system_readdir(dir) {
+    // Skip . and ..
+    if dirent.pointee.d_name.0 == 46
+         && (dirent.pointee.d_name.1 == 0
+               || (dirent.pointee.d_name.1 == 46
+                     && dirent.pointee.d_name.2 == 0)) {
+      continue
+    }
+
+    try body(dirent.pointee)
+  }
+}
+
+internal func recursiveRemove(
+  in dirfd: CInt,
+  path: UnsafePointer<CInterop.PlatformChar>
+) throws {
+  // First, deal with subdirectories
+  try forEachFile(in: dirfd, path: path) { dirent in
+    if dirent.d_type == SYSTEM_DT_DIR {
+      try withUnsafeBytes(of: dirent.d_name) {
+        try recursiveRemove(
+          in: dirfd,
+          path: $0.assumingMemoryBound(to: CInterop.PlatformChar.self)
+            .baseAddress!
+        )
+      }
+    }
+  }
+
+  // Now delete the contents of this directory
+  try forEachFile(in: dirfd, path: path) { dirent in
+    let flag: CInt
+
+    if dirent.d_type == SYSTEM_DT_DIR {
+      flag = SYSTEM_AT_REMOVE_DIR
+    } else {
+      flag = 0
+    }
+
+    let result = withUnsafeBytes(of: dirent.d_name) {
+      system_unlinkat(dirfd,
+                      $0.assumingMemoryBound(to: CInterop.PlatformChar.self)
+                        .baseAddress!,
+                      flag)
+    }
+
+    if result != 0 {
+      throw Errno.current
+    }
+  }
+}
+#endif
+
+internal let base64 = Array<UInt8>(
+  "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_".utf8
+)
+
+internal func makeDirectory(at: FilePath) throws -> Bool {
+  return try at.withPlatformString {
+    if system_mkdir($0, 0o700) == 0 {
```

Comment:
If we want this function to create directories with the hardwired mode 700, then it should probably carry a more specific name than just `makeDirectory`.